docs/build/guides/archival/extend-persistent-entry-js.mdx

```diff
@@ -0,0 +1,179 @@
+---
+title: Extend a persistent entry and contract using the JavaScript SDK
+description: How to extend a persistent entry and contract using the JavaScript SDK
+---
+
+Persistent storage is used for storing data and contracts that need to remain on the network for a long period.
+
+Each entry made to the persistent storage is stored as a key-value store. An entry consists of:
+
+- A key (identifier for the data)
+- A value (the data itself)
+- A Time-To-Live (TTL) that determines how long the data remains on the network
+
+When the TTL expires, the data is then archived on the network and is no longer accessible unless it is unarchived. To prevent data and contracts from being inaccessible, the TTL can be extended before it expires.
+
+## Extension Process
+
+1. Create the identifier according to what you're extending:
+
+- For a persistent entry, specify the contract address, storage key, and durability type (persistent)
+- For contract code, specify the contract ID
+
+2. Create a footprint and declare which storage locations will be extended and include the identifier for persistent entry, contract code, or both, depending on your intent.
+
+3. Build and submit the transaction:
+
+- Add the `extendFootprintTtl` operation with your intended TTL
+- Sign and submit the transaction
+
+:::note TTL can only be extended by the deployer or an authorized account. :::
+
+### Extending a persistent entry
+
+```javascript
+const StellarSdk = require("@stellar/stellar-sdk");
+const { Server } = require("@stellar/stellar-sdk/rpc");
+
+async function extendPersistentEntryTTL(contractId, storageKey, sourceKeypair) {
+  const server = new Server("https://soroban-testnet.stellar.org");
+  const account = await server.getAccount(sourceKeypair.publicKey());
+  const fee = "200100";
+
+  // Create an identifier for the entry
+  const persistentEntry = new StellarSdk.xdr.LedgerKey.contractData({
+    contract: StellarSdk.Address.fromString(contractId).toScAddress(),
+    key: StellarSdk.xdr.ScVal.scvSymbol(storageKey),
+    durability: StellarSdk.xdr.ContractDataDurability.persistent(),
+  });
+
+  // Create footprint for the entry
+  const footprint = new StellarSdk.xdr.LedgerFootprint({
+    readOnly: [],
+    readWrite: [persistentEntry],
```

Comment:
The Read/Write set must be empty.  The LedgerKey needs to be added to the readOnly footprint:

https://developers.stellar.org/docs/learn/encyclopedia/storage/state-archival#extendfootprintttlop